api/v1alpha1/shared_types.go

```diff
@@ -669,7 +669,14 @@ type CustomResponse struct {
 	ContentType *string `json:"contentType,omitempty"`
 
 	// Body of the Custom Response
-	Body CustomResponseBody `json:"body"`
+	//
+	// +optional
+	Body *CustomResponseBody `json:"body,omitempty"`
+
+	// Status Code of the HTTP response
+	//
+	// +optional
+	StatusCode *string `json:"statusCode,omitempty"`
```

Comment:
Nit: The `StatusCodeMatch` uses int for StatusCode, should we also use int here?

Reply:
oops, I'll just delete this field for now